### PR TITLE
refactor: tests for es package

### PR DESCRIPTION
**Pull request by kevinbheda**

### CODE REVIEW

File: dagger-core/src/main/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigBuilder.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package io.odpf.dagger.core.processors.external.es;
+
+import io.odpf.dagger.core.processors.common.OutputMapping;
+
+import java.util.Map;
+
+public class EsSourceConfigBuilder {
```
Comment (lavkesh) [inline dagger-core/src/main/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigBuilder.java:7]: do we really need this class? It's only being used in the test? If we need this we can move it to test package?

Comment (kevinbheda) [reply dagger-core/src/main/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigBuilder.java:7]: Yes we need this class, it makes the code much more readable. given in this pr we are focusing on refactoring tests, wanted to make minimal changes to actual code. will start using the builder class in main code afterwards.

File: dagger-core/src/test/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigTest.java [hunk 2]
```diff
@@ -198,29 +229,53 @@ public void shouldReturnMandatoryFields() {
...
+        EsSourceConfigBuilder validEsSourceConfigBuilder = getValidEsSourceConfigBuilder();
+        EsSourceConfig esSourceConfig = validEsSourceConfigBuilder
+                .setOutputMapping(Collections.emptyMap())
+                .createEsSourceConfig();
         esSourceConfig.validateFields();
     }
 
     @Test
     public void shouldValidateWhenEndpointPatternIsEmpty() {
         expectedException.expect(IllegalArgumentException.class);
```
Comment (gauravsinghania) [inline dagger-core/src/test/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigTest.java:252]: can we change the exception to argument captor?

Comment (kevinbheda) [reply dagger-core/src/test/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigTest.java:252]: thanks for pointing out, had upgraded the junit version afterwards so missed changing expected rule to asserthrows here , will update it

**Pull request merged**